### ISSUE TITLE
Modularize code for various protocols

### ISSUE THREAD

**Issue by pgmsft** [Microsoft/ethr#24]
The code needs to be modularized to have interface for protocols and handlers for running Bandwidth/Latency/CPS/PPS tests for those protocols. Right now the code is not using clean abstractions.

For example, we should have something like:
type ethrProto interface{
func runBandwidthTest(param…)
func runLatencyTest(param…)
}

type ethrProtoTcp struct {
}

func (p *ethrProtoTcp) runBandwidth(param…) {
}

**Comment by piotrzarycki** [comment]
Ok, i can handle it :)

**Comment by pgmsft** [comment]
I cleaned up the code a little bit already and given that server starts listening on all the ports in parallel, using interface was not helping.

Please see if you can pick up another issue like UDP latency testing.
